src/pal/src/thread/process.cpp

```diff
@@ -2971,58 +2971,74 @@ PROCAbortInitialize()
 
 /*++
 Function:
-  PROCAbort()
+  PROCCreateCrashDumpIfEnabled
 
-  Aborts the process after calling the shutdown cleanup handler. This function
-  should be called instead of calling abort() directly.
-  
-  Does not return
+  Creates crash dump of the process (if enabled). Can be
+  called from the unhandled native exception handler.
+
+(no return value)
 --*/
-PAL_NORETURN
 VOID
-PROCAbort()
+PROCCreateCrashDumpIfEnabled()
 {
-    // Do any shutdown cleanup before aborting or creating a core dump
-    PROCNotifyProcessShutdown();
-
 #if HAVE_PRCTL_H && HAVE_PR_SET_PTRACER
     // If enabled, launch the create minidump utility and wait until it completes
-    if (g_argvCreateDump[0] != nullptr)
-    {
-        // Fork the core dump child process.
-        pid_t childpid = fork();
+    if (g_argvCreateDump[0] == nullptr)
+        return;
+
+    // Fork the core dump child process.
+    pid_t childpid = fork();
 
-        // If error, write an error to trace log and abort
-        if (childpid == -1)
+    // If error, write an error to trace log and abort
+    if (childpid == -1)
+    {
+        ERROR("PROCAbort: fork() FAILED %d (%s)\n", errno, strerror(errno));
```

Comment:
A nit - the messages were not updated to reflect the fact that they are now in a new function (they still refer to PROCAbort).